### PR TITLE
vm/permissions

### PR DESCRIPTION
**Pull request by harshshah-tenable**
- Changes for Permissions API

### CODE REVIEW

File: tenable/io/v3/vm/permissions/api.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    _schema = PermissionSchema()
+
+    def change(self, otype: str, id: int, *acls: List) -> None:
+        '''
+        Modify the permission of a specific object.
+
+        :devportal:`permissions: change <permissions-change>`
+
+        Args:
+            otype:
```
Comment (parvdesai-tenable) [inline tenable/io/v3/vm/permissions/api.py:36]: arg type here

File: tenable/io/v3/vm/permissions/api.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            :obj:`list`:
+                The permission recourse record listings for the
+                specified object.
+
+        Examples:
+            >>> tio.v3.vm.permissions.list(otype, id)
+        '''
+        return self._get(f'{otype}/{id}')['acls']
+
+    def search(self):
```
Comment (parvdesai-tenable) [inline tenable/io/v3/vm/permissions/api.py:85]: shouldn't this be updated to  `def search(self, **kwargs):`??

**Review by parvdesai-tenable — APPROVED** [review]

File: tenable/io/v3/vm/permissions/api.py (hunk 1 above)
Comment (harshshah-tenable) [reply tenable/io/v3/vm/permissions/api.py]: Updated

File: tenable/io/v3/vm/permissions/api.py (hunk 2 above)
Comment (harshshah-tenable) [reply tenable/io/v3/vm/permissions/api.py]: updated

File: tenable/io/v3/vm/permissions/schema.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class PermissionAclSchema(Schema):
+    '''
+    Schema for acls for change permissions action
+    '''
+    type = fields.Str(required=True, validate=validate.OneOf(
+        ['default', 'user', 'group']))
+    id = fields.Int(required=True)
+    permissions = fields.Int(required=True)
```
Comment (dhvanilshah-tenable) [inline tenable/io/v3/vm/permissions/schema.py:14]: @harshshah-tenable Have you verified if both of them are required params? The spec here doesn't mention it as required.

File: tenable/io/v3/vm/permissions/api.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+'''
+from typing import Dict, List
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+from tenable.io.v3.vm.permissions.schema import PermissionSchema
+
+
+class PermissionsAPI(ExploreBaseEndpoint):
+    '''
+    This will contain all methods related to Permissions API
```
[the remark this answers is not in the document]
Comment (harshshah-tenable) [reply tenable/io/v3/vm/permissions/api.py]: updated

File: tests/io/v3/vm/permissions/test_permissions.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
+'''
```
Comment (parvdesai-tenable) [inline tests/io/v3/vm/permissions/test_permissions.py:1]: Change filename to `test_permissions_api.py`
Reference: https://github.com/tenable/pyTenable/pull/502/files

File: tests/io/v3/vm/permissions/test_schema.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+'''
```
Comment (parvdesai-tenable) [inline tests/io/v3/vm/permissions/test_schema.py:1]: Change filename to `test_permissions_schema.py`
Reference: https://github.com/tenable/pyTenable/pull/502/files

File: tenable/io/v3/vm/permissions/schema.py (hunk 3 above)
Comment (harshshah-tenable) [reply tenable/io/v3/vm/permissions/schema.py]: Yes validated again these are not required params in payload for API removing the required keyword from schema

File: tests/io/v3/vm/permissions/test_schema.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+from tenable.io.v3.vm.permissions.schema import PermissionSchema
+
+
+def test_schema():
+    schema = PermissionSchema()
+    payload = schema.dump(schema.load(
+        {'acls': [{'type': 'user', 'id': '2236706', 'permissions': 64}]}))
+    assert payload == {
+        'acls': [{'type': 'user', 'id': 2236706, 'permissions': 64}]}
```
Comment (parvdesai-tenable) [inline tests/io/v3/vm/permissions/test_schema.py:12]: We can make this object as a constant and avoid repetation

**Review by parvdesai-tenable — APPROVED** [review]

File: tests/io/v3/vm/permissions/test_schema.py (hunk 7 above)
Comment (harshshah-tenable) [reply tests/io/v3/vm/permissions/test_schema.py]: Updated

File: tests/io/v3/vm/permissions/test_schema.py (hunk 6 above)
Comment (harshshah-tenable) [reply tests/io/v3/vm/permissions/test_schema.py]: done

File: tests/io/v3/vm/permissions/test_permissions.py (hunk 5 above)
Comment (harshshah-tenable) [reply tests/io/v3/vm/permissions/test_permissions.py]: done

**Review by dhvanilshah-tenable — APPROVED** [review]

**Review by tomeara — APPROVED** [review]

**Pull request merged**